lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -417,135 +424,149 @@ void HWMemSimImpl::generateMemory(HWModuleOp op, FirMemory mem) {
 
   // Add logic to initialize the memory and any internal registers to random
   // values.
-  constexpr unsigned randomWidth = 32;
-  sv::RegOp randomMemReg;
-  b.create<sv::IfDefOp>("SYNTHESIS", std::function<void()>(), [&]() {
-    sv::RegOp randReg;
-    StringRef initvar;
-
-    // Declare variables for use by memory randomization logic.
-    b.create<sv::IfDefOp>("RANDOMIZE_MEM_INIT", [&]() {
-      initvar = moduleNamespace.newName("initvar");
-      b.create<sv::VerbatimOp>("integer " + Twine(initvar) + ";\n");
-      randomMemReg = b.create<sv::RegOp>(
-          b.getIntegerType(llvm::divideCeil(mem.dataWidth, randomWidth) *
-                           randomWidth),
-          b.getStringAttr("_RANDOM_MEM"),
-          b.getStringAttr(moduleNamespace.newName("_RANDOM_MEM")));
-    });
+  if (!disableMemRandomization || !disableRegRandomization) {
```

Comment:
Can the PR decrease the nesting by inverting the condition and early exiting or by refactoring the randomization logic into a method?

Reply:
Good call, looks like code after the conditional can be safely hoisted before, so can move to early-exit.  Thanks!